core/network/src/protocol/sync.rs

```diff
@@ -265,7 +305,10 @@ impl<B: BlockT> ChainSync<B> {
 				protocol.disconnect_peer(who);
 			},
 			(Ok(BlockStatus::Unknown), _) if self.queue_blocks.len() > MAJOR_SYNC_BLOCKS => {
-				// when actively syncing the common point moves too fast.
+				// Here we have more than `MAJOR_SYNC_BLOCKS` in the import queue which means that
+				// we are actively downloading, the common ancestor block is moving too fast to do
+				// an ancestor search and so we assume the best queued block is the common point.
+				// TODO: WHY?
```

Comment:
Why do we not do ancestor search if we're more than 5 blocks behind? Just pointless as we'll have to re-do it once those are imported anyway?

Reply:
I assume it's an heuristic that probably by the point those blocks are imported we'll be on the same chain (or closer to one another then we are now anyway), so it's better to just delay the ancestor search.

Reply:
Yep